src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
     public function canFilterAnyColumns($gridField)
     {
         $list = $gridField->getList();
-
-        if (!$this->checkDataType($list)) {
+        if (!($list instanceof Filterable) || !$this->checkDataType($list)) {
             return false;
         }
-
-        $columns = $gridField->getColumns();
-        foreach ($columns as $columnField) {
-            $metadata = $gridField->getColumnMetadata($columnField);
-            $title = $metadata['title'];
-
-            if ($title && $list->canFilterBy($columnField)) {
-                return true;
+        $modelClass = $gridField->getModelClass();
+        $searchableFields = Config::inst()->get($modelClass, 'searchable_fields');
+        if (!empty($searchableFields)) {
+            // Scenario #1 - searchable_fields is defined on model class
+            foreach ($searchableFields as $searchableField) {
+                if ($list->canFilterBy($searchableField)) {
+                    return true;
+                }
+            }
```

Comment:
`canFilterBy()` is fine for `summary_fields()` - but we can declare `searchable_fields` that aren't _traditionally_ filterable here, and then tell `SearchContext` how to search using those fields. If a field is in searchable fields that _isn't_ in summary fields, we should just assume it's filterable and let `SearchContext` sort it out.

Reply:
Rewrote the method